src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -376,6 +379,26 @@ void buildPrimaryKeyConfiguration(
 
         const ASTDictionaryAttributeDeclaration * dict_attr = (*it)->as<const ASTDictionaryAttributeDeclaration>();
 
+        auto key_type = DataTypeFactory::instance().tryGet(dict_attr->type);
+
+        auto check_sample_dict_key_is_correct = context->getSettings().check_sample_dict_key_is_correct;
```

Comment:
It's better to use `getSettingsRef` because `getSettings` makes a copy of the `Settings` struct which is several KBs.

Reply:
ok, thank you for the advice.